packages/compass-explain-plan/src/modules/explain.js

```diff
@@ -406,6 +415,22 @@ export const changeExplainPlanState = (explainState) => {
   };
 };
 
+/**
+ * Sets the explain plan state to `requested`
+ */
+export const startExplainPlan = () => {
+  return (dispatch) => {
+    const abortController = new AbortController();
+    return dispatch(explainStateChanged(EXPLAIN_STATES.REQUESTED, abortController));
+  };
+};
+
+export const cancelExplainPlan = () => {
+  return (dispatch, getStore) => {
+    getStore().explain.abortController?.abort();
```

Comment:
Because you are aborting without specifying a `reason` and not handling cancellation as some special case as soon as you merge this branch with main (which updates electron), you'll start getting a pretty garbage message. I would suggest to either explicitly handle abort state as [we do in aggregation plugin](https://github.com/mongodb-js/compass/blob/e4dda02d4c291f949fd862ac0347beb1b12f86d7/packages/compass-aggregations/src/modules/aggregation.ts#L235-L251) or at least explicitly provide an error here to make sure that the message you will get in UI is controlled by the code here, [similar to compass-crud](https://github.com/mongodb-js/compass/blob/c6cf421f51c55178ef005cec77dcc0fc37494b4a/packages/compass-crud/src/stores/crud-store.ts#L1395)

Also as we are keeping the controller in the store, it would be nice to clean it up after it was used

Reply:
I have a PR that would probably simplify that a bit already, moving the tree-stages.js out of the store, so the store would only need the `explain`, would it make sense to merge this and do the change in a follow-up? I'd be in favor of cleaning up explain.js, seems unnecessarily convoluted for a store that at the moment have only one action.

Reply:
If you meant merging your PR first and then adapting this PR accordingly (if needed) then I am good that.

Reply:
User cancelling the explain is now a more explicit action - and restores the previous state (which ever it was).